account_invoice_cancel_with_payment_order/wizards/validate_invoice_cancel.xml

```diff
@@ -0,0 +1,29 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!-- Copyright 2019 ACSONE SA/NV
+     License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl). -->
+
+<odoo>
+
+    <record model="ir.ui.view" id="validate_invoice_cancel_form_view">
+        <field name="name">validate.invoice.cancel.form (in account_invoice_cancel_with_payment_order)</field>
+        <field name="model">validate.invoice.cancel</field>
+        <field name="arch" type="xml">
+            <form string="Validate Invoice Cancel">
+                <group>
+                   <p>WARNING ! You try to cancel an invoice related to a payment order.</p>
+                </group>
+                 <group groups="account.group_account_manager">
+                   <p>This process can be dangerous. You assume the possible consequences of this action.</p>
```

Comment:
It would be nice to give a hint about the consequences, otherwise it's a bit stressful to be a user :)

Also, I think the behaviour must be different depending on the state of the payment order.
Indeed, if the linked payment line is in a payment order that is not sent, I tend to think the consequence is minimal.

If the payment line is in a payment order that has not been sent, the payment line is not reconciled with the move line and when the invoice is cancelled, and later the payment sent, the invoice will not be reconciled, so in that case I'd say it should be forbidden to cancel the invoice, and the user must first remove it from the payment order.